src/test/func/malloc/malloc.cc

```diff
@@ -138,6 +138,22 @@ void test_memalign(size_t size, size_t align, int err, bool null)
   check_result(size, align, p, err, null);
 }
 
+#if !defined(SNMALLOC_NO_REALLOCARR)
+void test_reallocarr(void* p, size_t nmemb, size_t size, int err, bool null)
+{
+  size_t old_size = 0;
+  if (p != nullptr)
+    old_size = our_malloc_usable_size(p);
+
+  printf("reallocarr(%p(%zu), %zu)\n", p, old_size, nmemb * size);
+  errno = SUCCESS;
+  int r = our_reallocarr(&p, nmemb, size);
+  if (r != SUCCESS && size != 0)
```

Comment:
Drop `size != 0` I don't think the spec requires `size == 0` to deallocate.  It doesn't for `realloc`, so I assume it doesn't here.